src/api/app/components/button_component.html.haml

```diff
@@ -0,0 +1,3 @@
+%button.btn{ class: css_style, data: @button_data }
+  = icon
```

Comment:
I simplified the logic and the structure of the component, getting rid of the `slots` because they were not really needed for such a simple component.